cmd/loadtest/loadtest.go

```diff
@@ -347,32 +351,48 @@ func runLoadTest(ctx context.Context) error {
 		overallTimer = new(time.Timer)
 	}
 
+	// // Dial the Ethereum RPC server.
 	rpc, err := ethrpc.DialContext(ctx, *inputLoadTestParams.RPCUrl)
 	if err != nil {
 		log.Error().Err(err).Msg("Unable to dial rpc")
 		return err
 	}
+	defer rpc.Close()
 	rpc.SetHeader("Accept-Encoding", "identity")
 	ec := ethclient.NewClient(rpc)
 
+	// Define the main loop function.
+	// Make sure to define any logic associated to the load test (initialization, main load test loop
+	// or ven completion steps) in this function in order to handle cancellation signals properly.
 	loopFunc := func() error {
-		err = initializeLoadTestParams(ctx, ec)
-		if err != nil {
+		if initializeLoadTestParams(ctx, ec) != nil {
+			log.Error().Err(err).Msg("Error initializing load test parameters")
 			return err
 		}
 
-		return mainLoop(ctx, ec, rpc)
+		if mainLoop(ctx, ec, rpc) != nil {
+			log.Error().Err(err).Msg("Error during the main load test loop")
+			return err
+		}
+
+		if completeLoadTest(ctx, ec, rpc) != nil {
+			log.Error().Err(err).Msg("Encountered error while wrapping up loadtest")
+		}
+		return nil
```

Comment:
why not return complete load test `err`?

Reply:
because I wanted to display a custom error but I can do this yep